### PR TITLE
Allow configuration of dynamodb storage to specify the max retries of aws sdk

### PR DESCRIPTION
**Pull request by rin1221**
This was an issue if you are wanting errors to manifest themselves in vault as the errors weren't being shown. By setting the variable to 0 you are able to get error messages from the sdk calls if it fails once and then the clients can handle the retry logic.

### CODE REVIEW

**Comment by jefferai** [comment]
AWS has a default retry value; if the config value isn't set it should use the AWS default, not zero.

**Comment by rin1221** [comment]
I agree. I will update the branch to reflect that.

File: physical/dynamodb/dynamodb.go [hunk 1]
```diff
@@ -183,6 +183,18 @@ func NewDynamoDBBackend(conf map[string]string, logger log.Logger) (physical.Bac
 		}
 	}
 
+	dynamodbMaxRetryString := os.Getenv("AWS_DYNAMODB_MAX_RETRIES")
+	if dynamodbMaxRetryString == "" {
+		dynamodbMaxRetryString = conf["dynamodb_max_retries"]
+		if dynamodbMaxRetryString == "" {
+			dynamodbMaxRetryString = "-1"
```
Comment (jefferai) [inline physical/dynamodb/dynamodb.go]: Please use the aws sdk const name, not the value.

**Comment by jefferai** [comment]
Thanks!

**Pull request merged**